migration-engine/connectors/sql-migration-connector/src/flavour.rs

```diff
@@ -253,11 +253,19 @@ pub(crate) trait SqlFlavour:
         &'a mut self,
         migrations: &'a [MigrationDirectory],
         shadow_database_url: Option<String>,
+        namespaces: Option<Namespaces>,
     ) -> BoxFuture<'a, ConnectorResult<SqlSchema>>;
 
     /// Receive and validate connector params.
     fn set_params(&mut self, connector_params: ConnectorParams) -> ConnectorResult<()>;
 
+    /// Sets the preview features. This is currently useful for MultiSchema, as we want to
+    /// grab the namespaces we're expected to diff/work on, which are generally set in
+    /// the schema.
+    /// WARNING: This may silently not do anything if the connector is in the initial state.
+    /// If this is ever a problem, considering returning an indicator of success.
```

Comment:
:+1:

I think we'll probably want to pop preview features out of the `Params` part of the connector state in a separate PR. It's fine to have somewhat fuzzy provenance for preview features: if the user manifested the intent to turn on a preview feature _anywhere_ (CLI arguments, Prisma schema), we can always keep adding to the preview features set from multiple places.

Reply:
We should maybe also consider crashing loudly if `set_preview_feature()` is called on a connector that is in `State::Initial`, since that should never happen.

Reply:
From zoom discussion: we should log with `warn!()`.

Reply:
Issue: https://github.com/prisma/schema-team/issues/366